#27 Refactor(App): Refactoring django structure

uridonget opened the pull request:
# refactor(App) \#27
## Overview
장고의 account 앱에 기능이 과도하게 집중되어서 효율적인 개발에 좋지 못한 영향을 주고 있었습니다. 
account 앱에서 기능 단위로 앱을 여러개로 나누어 가독성을 높이고 유지보수를 용이하게 하였습니다.

<!---- Resolves: #(Isuue Number) -->
## PR Type
account 앱의 기능을 나누어
account, authentication(oauth & local_auth), follow 을 만들었습니다. 보다 직관적으로 어떤 기능에 사용되는 앱인지 개발환경에서 확인할 수 있습니다.

- [x] Chore(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] refactoring

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.

uridonget committed:
#23 Refactor(Utils): 프로젝트 전체에서 사용할 util 파일 생성

core 디렉토리의 utils.py 파일을 생성하였음.

프로젝트 전체에서 사용할 함수들을 utils.py 파일에 작성할 것.

JWT 인증 함수인 authenticate_token 함수를 utils.py 파일로 이동하였음.

uridonget committed:
#23 Refactor(Follows): 팔로우 관리를 위한 앱을 생성

유저의 팔로우 관리를 위하여 follows 앱을 생성하였음.

Follows 앱으로 데이터베이스 Follow 모델을 이동하였음.

유저의 팔로우 관리에 사용되는 함수들을 follows 앱의 views.py 파일로 이동하였음.

관련 endpoint를 새로운 follow 앱에 맞추어 그대로 이동시키거나 소폭 수정하였음. (API 명세서에 수정사항 반영할 것)

uridonget committed:
#23 Chore(Follow): 앱 이름 수정

기존의 'follows' 앱 이름을 'follow' 으로 변경하였음.

기존의 'get_follows' 함수명을 'get_follower_list' 으로 변경하였음.

uridonget committed:
#23 Refactor(Authentication): 인증 앱 생성

기존의 'account' 앱에 포함되어 있었던 인증과정을 새로운 앱에 이동하였음.

authentication 앱 디렉토리 아래에 oauth 디렉토리와 local_auth 디렉토리를 생성하였음.

해당 디렉토리 아래에 views.py 파일과 urls.py 파일을 생성하여 해당되는 함수와 API 엔드포인트를 할당하여 코드의 가독성과 유지보수를 용이하게 하였음.

uridonget committed:
#23 Chore(Profile): 프로필 이미지 삭제 로직 수정

이전 이미지와 새로운 이미지의 확장자가 다를 때, 이전 이미지가 삭제되지 않는 오류를 발견하여 수정하였습니다.

allorak333 commented on backend/account/views.py:
```diff
@@ -142,17 +142,23 @@ def change_profile_image(request):
-                )
-
         saved_path = default_storage.save(file_path, ContentFile(image.read()))
         image_url = f"/media/{saved_path}"
```
저도 이미지 확장자가 다르게 오는 상황을 생각해보진 못했네요, 다만 이전에 user.imagePath 에 서버 주소를 담지 않아 발생하는 이슈(다른 서버에 있는 이미지 사용 시, 프론트 엔드 코드의 로직 불일치)와 동일한 문제가 발생 할 것 같습니다. 서버 주소를 추가하는 로직이 필요해보입니다.

uridonget replied:
디테일을 신경쓰지 못했습니다! 피드백에 맞추어 수정하였습니다!

uridonget committed:
#23 Chore(Profile): 프로필 이미지 저장 및 삭제 로직 수정

유저의 imagePath에 프로필 이미지 url을 저장할 때, 내부경로 뿐 아니라 프론트엔드 서버 url를 추가하도록 하였음.

위의 수정사항으로 인하여 기존의 프로필 이미지 삭제 로직을 수정하였음.

uridonget committed:
#23 Chore(Session): 세션 관련 함수에서 상태코드 수정

authenticate_code, get_local_auth_token 함수에서 '서버의 세션에 인증코드가 없는 경우'와 '잘못된 인증코드를 제출한 경우'에
동일한 상태코드(400)을 반환하는 것을 확인.

서버의 세션에 인증코드가 없는 경우에는 다시 이메일로 인증코드를 발송해야하고, 잘못된 인증코드를 제출한 경우에는 다시 입력값을 받아야하기에 상태코드를 다르게 설정하는 것이 유리함.

서버의 세션에 인증코드가 존재하지 않으면 404(Not found) 상태코드를 반환하도록 수정하였음.

allorak333 reviewed (approved):
LGTM!
